From c87dc5e49a6daf64d3cf73402dcaf0dba1f4555f Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Tue, 24 May 2016 16:49:28 +0300
Subject: [PATCH 0235/1345] net: mvpp2x: Add cell_index for the device, from
 dts or increasing index

commit  fe5f7d6821171e8a3bb96e5163a6b291841dd76d from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

-Add cell_index for the device, from dts or increasing index

Change-Id: Ia931a26fef202ab9ee5465fe822e1c799df027bf
Reviewed-on: http://vgitil04.il.marvell.com:8080/29941
Reviewed-by: Yuval Caduri <cyuval@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c |   15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
index 2de87c1..16fccb1 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
@@ -4411,6 +4411,8 @@ static int mv_pp2x_platform_data_get(struct platform_device *pdev,
 		struct mv_pp2x *priv,	u32 *cell_index, int *port_count)
 {
 	struct mv_pp2x_hw *hw = &priv->hw;
+	static int auto_cell_index;
+	static bool cell_index_dts_flag;
 	const struct of_device_id *match;
 #if !defined(CONFIG_MV_PP2_FPGA) && !defined(CONFIG_MV_PP2_PALLADIUM)
 	struct device_node *dn = pdev->dev.of_node;
@@ -4431,8 +4433,16 @@ static int mv_pp2x_platform_data_get(struct platform_device *pdev,
 	priv->pp2xdata = (struct mv_pp2x_platform_data *) match->data;
 
 #if !defined(CONFIG_MV_PP2_FPGA) && !defined(CONFIG_MV_PP2_PALLADIUM)
-	if (of_property_read_u32(dn, "cell-index", cell_index))
-		*cell_index = 0;
+	if (of_property_read_u32(dn, "cell-index", cell_index)) {
+		*cell_index = auto_cell_index;
+		auto_cell_index++;
+	}
+
+	else
+		cell_index_dts_flag = true;
+
+	if (auto_cell_index && cell_index_dts_flag)
+		return -ENXIO;
 
 	MVPP2_PRINT_VAR(*cell_index);
 
@@ -4608,6 +4618,7 @@ static int mv_pp2x_platform_data_get(struct platform_device *pdev,
 	/* Get system's tclk rate */
 	hw->tclk = clk_get_rate(hw->pp_clk);
 	MVPP2_PRINT_VAR(hw->tclk);
+
 #else
 	hw->tclk = 25000000;
 #endif
-- 
1.7.9.5

